docs/internal/interop.md

```diff
@@ -0,0 +1,351 @@
+# Verona Interoperability / Sandbox
+
+This is an initial attemp at mapping all the requirements that the
+interoperability layer will have in Verona to use external code written in other
+languages (like C++, Rust, etc) in a safe manner (sanbdox).
+
+## Usage in Verona
+
+The aim is to have external code in Verona with a native interface that looks
+and feels like Verona code. We want to abstract away the entirety of the sandbox
+functionality, the foreign function dispatch and serialisations, the execution
+models, etc.
+
+The syntax is still largely undefined, but this is an example of using the C++
+`std::vector` class inside Verona as pseudo-Verona code:
+
+```c
+// include/import the c++ code as a Verona module
+type StdVec = "c++::vector";
```

Comment:
Is this the latest version of Verona import syntax?

We should never write code that names C++ types as anything other than Verona types that are nested inside the class (package) the encapsulates a specific C++ library.  Something like:

```verona
import foreign_lib = some.package.foreign;
```

That will live in our filesystem as `some/package/foreign`, which will contain a `verona_interop.json` (or some better name), which will include everything required to generate verona types (C++ header, library, compilation flags).  If that header includes `std::vector` then `foreign_lib::std::vector` is a valid Verona type name and we can instantiate it (though its constructor, as exposed to Verona, is defined only on sandboxed regions).

Reply:
I think the key thing here is that there is no special syntax for 'this is a foreign class' at the import site.  There is just syntax (whatever it may be) for importing a module.  From the world outside of that module, it looks just like a Verona module, though the nested classes inside it have constraints on the kind of region that they can be created inside (i.e. only sandboxed regions, possibly only a specific kind of sandbox region, depending on how the code is provided to Verona).  We have an explicit design decision to not add any custom syntax for foreign code.